### DISCUSSION TITLE
ml:source at the ml:RecordSet level is not used

### DISCUSSION THREAD

**Discussion by marcenacp** [mlcommons/croissant#142] [Ideas]
[`ml:source`](https://github.com/mlcommons/croissant/blob/main/docs/croissant-spec.md#source) is a field for both RecordSets and Fields.

The source at the RecordSet level is not used by the machine, as sources at the Field level prevail. The source also doesn't increase readability for the human, which either a) know the JSON-LD/Croissant standard, or b) interact through a visualizer/wizard.

I propose to drop ml:source at the RecordSet level, as it creates noise in the syntax. WDYT?

**Result of the discussion with @benjelloun:**

- For the moment, we can manage all cases by declaring the `source` at the Field level.
- Nevertheless, declaring the `source` at the RecordSet level is an interesting approach. Notably, it allows to clearly distinguish left/right in joins. However, this approach still has open questions:
  - A lot of RecordSets (at least in our example datasets) do not have one "source" but several sources that are joined. How to clearly account for this?
  - How to "chain" dataExtraction from the RecordSet to the Field? (See for example `caption_annotations-files/annotations` in COCO2014.)

**Comment by benjelloun** [comment]
One issue with the current approach is that specifying transformations that apply to the entire recordset is a bit wonky, since they have to be repeated on each Field, and the correspondence between fields in the same row is implicit.
